frontend/resetPassword.html

```diff
@@ -0,0 +1,104 @@
+      }
+      form {
+        display: flex;
+        flex-direction: column;
+      }
+      input {
+        margin: 10px 0;
+        padding: 8px;
+      }
+      button {
+        margin-top: 10px;
+        padding: 8px;
+        background-color: #007bff;
+        color: white;
+        border: none;
+        cursor: pointer;
+      }
+      #message {
+        margin-top: 20px;
+        text-align: center;
+      }
+    </style>
+  </head>
+  <body>
+    <div class="container">
+        <h2>Reset Password</h2>
+        <form id="resetForm">
+            <input type="password" id="newPassword" placeholder="New Password" required>
+            <input type="password" id="confirmPassword" placeholder="Confirm New Password" required>
+            <button type="submit">Reset Password</button>
+        </form>
+        <div id="message"></div>
+    </div>
+    <script>
+        const API_URL = "http://localhost:5000/api";
+        const form = document.getElementById("resetForm");
+        const messageDiv = document.getElementById("message");
+
+        form.addEventListener("submit", async (e) => {
+            e.preventDefault();
+            const newPassword = document.getElementById("newPassword").value;
+            const confirmPassword = document.getElementById("confirmPassword").value;
+
+            if (newPassword !== confirmPassword) {
+                messageDiv.textContent = "Passwords do not match. Please try again.";
+                return;
+            }
+
+            // Extract token from URL
+            const token = window.location.pathname.split("/reset-password/")[1];
+            console.log("Token extracted from URL:", token); // Add this line for debugging
+
+            if (!token) {
+                messageDiv.textContent = "Invalid or missing reset token. Please request a new password reset.";
+                return;
+            }
+
+            try {
+                const response = await fetch(`${API_URL}/auth/reset-password/${token}`, {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                    body: JSON.stringify({ password: newPassword }),
+                });
+
+                if (response.ok) {
+                    messageDiv.textContent = "Password reset successfully. You can now log in with your new password.";
+                    setTimeout(() => {
+                        window.location.href = "signin.html";
+                    }, 3000);
+                } else {
+                    const errorData = await response.text();
+                    messageDiv.textContent = `Password reset failed: ${errorData}`;
+                }
+            } catch (error) {
+                console.error("Error:", error);
+                messageDiv.textContent = "An error occurred. Please try again.";
+            }
+        });
+    </script>
+</body>
+</html>
```

Comment:
Move CSS and JavaScript to separate files to improve maintainability, especially as the project grows.
Create a styles.css file for the CSS rules.
Create a reset-password.js file for JavaScript.
Provide an option to toggle the visibility of the passwords
The code is functional and meets most basic requirements. The recommendations aim to improve security, accessibility, and user experience.

Reply:
The HTML is well-organized, making it easy to follow the layout and functionality. The use of <div> for the container, a form for input, and a dedicated message area is clear and semantic.
The styles provide a clean and modern look, making the form visually appealing. The use of flexbox for centering elements on the page is effective.
improvements;
Password Complexity Requirements: Consider adding validation for password complexity (e.g., minimum length, special characters) to enhance security. This can be done in the JavaScript section.